plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/ProxySettings.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.s3;
+
+import com.amazonaws.Protocol;
+import org.opensearch.common.Strings;
+import org.opensearch.common.settings.SettingsException;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.Objects;
+
+public class ProxySettings {
+    public static final ProxySettings NO_PROXY_SETTINGS = new ProxySettings(ProxyType.DIRECT, null, -1, null, null);
+
+    public static enum ProxyType {
+        HTTP(Protocol.HTTP.name()),
+        HTTPS(Protocol.HTTPS.name()),
+        SOCKS("SOCKS"),
+        DIRECT("DIRECT");
+
+        private final String name;
+
+        private ProxyType(String name) {
+            this.name = name;
+        }
+
+        public Protocol toProtocol() {
+            if (this == DIRECT) {
+                // We check it in settings,
+                // the probability that it could be thrown is small, but how knows
+                throw new SettingsException("Couldn't convert to S3 protocol");
+            } else if (this == SOCKS) {
+                throw new SettingsException("Couldn't convert to S3 protocol. SOCKS is not supportedS3");
```

Comment:
```suggest
SOCKS is not supported by S3
```

Reply:
This code convert `ProxyType` to AWS Protocol objects, whic does not support SOCKS. SOCKS implementation was done by extending SSLSocketFactory

Reply:
Fix the typo in the message? `supportedS3`